Figgle.Generator/FiggleSourceGenerator.cs

```diff
@@ -104,6 +106,9 @@ public void Execute(GeneratorExecutionContext context)
     {
         context.AddSource("GenerateFiggleTextAttribute.cs", AttributeSource);
 
+        var additionalFiles = context.AdditionalFiles.ToImmutableDictionary(
```

Comment:
Constructing a dictionary this way will throw if more than one path has the same file name.

Filtering to include only `.flf` files will help, but there'd still be a runtime error if the user had the equivalent of:

```xml
<ItemGroup>
    <AdditionalFile Include="Foo.flf" />
    <AdditionalFile Include="Folder\Foo.flf" />
</ItemGroup>
```

Rather than throwing in this case, I think we should use `context.ReportDiagnostic` to report a warning for this scenario, so the user knows what went wrong.

Another option is to require the full path to the item (e.g. `Foo` or `Folder\Foo`).

Reply:
I removed the dictionary construction and will iterate the `AdditionalFiles` array directly as suggested in the other comment.